packages/components/pagination/src/pagination.tsx

```diff
@@ -239,40 +243,46 @@ export default function Pagination({
     return pages
   }
 
+  if (hideOnSinglePage && totalPages <= 1) {
+    return null
+  }
```

Comment:
I think we shouldn't return null inside the pagination. As it will trigger all unnecessary useEffect